accepted/00101-health-check-tool.md

```diff
@@ -0,0 +1,290 @@
+    * Recommendation Logic: For recommendations, the alerting mechanism can be extended with additional logic to suggest actions.
+
+
+## Metrics
+
+* Server status and resource utilization
+
+* Database health and integrity
+
+* Status of relevant Uyuni services
+
+
+## Saline vs Health Check Tool
+
+We want to differentiate conceptually between "Saline" and the "Health Check Tool". In this regard, these are two different components with different purposes.
+
+### Saline: Salt state application monitoring to be integrated to the product.
+Saline exposes Salt state metrics to Prometheus. It is a tool meant to attach to Salt Master sockets and analyze and extract Salt state/jobs metrics from a live and running Uyuni server.
+
+Saline setup is done is two steps:
+
+1. Install `saline` RPM in the Uyuni server. This provides "setup" script and new "Formulas with Forms".
+2. Run `saline-setup run` to configure and enable the `salined` service to attach to the Salt Master.
+3. In the "Formulas with Forms" UI, new formulas will appear: "Saline Prometheus" and "Saline Grafana". These can be used to automatically configure your Prometheus and Grafana instances to get the metrics and dashboards from Saline.
+
+Saline already provides an easy way to integrate with the current Monitoring Stack (Formulas with Forms).
+
+The `salined` service must be able to attach to the Salt Master. At the moment this is done by connecting to Salt Master IPC sockets, which means `salined` needs to live in the same system/container than the Salt Master.
+
+Saline should be configurable to use IPC or TCP sockets. Switching Salt Master to use TCP sockets would enable Saline to run on a different container than the Salt Master.
+
+The Saline engine running on the Uyuni server, will provide data and metrics to feed Prometheus and Grafana instances and also a future UI with a live view of running actions. Eventually, this component should be integrated as part of the default Uyuni server stack.
+
+
+### Health Check Tool: An standalone tool you can run on an running Uyuni server or use in a disconnected setup (via supportconfig)
+In this case, this tool is mean not only to provide a picture of the current health status of a live running the Uyuni server, where the tool has access to it and can fetch data in real time, but it is meant to also help engineers and supporters to analyze and debug problems on disconnected setups, where the tool doesn't have access to an actual running Uyuni server, but only has a "supportconfig" as a source of data.
+
+In this regard, on disconnected setups, the tool cannot rely on existing Uyuni server components, neither existing Prometheus and Grafana instances, but it should be able to deploy its own instances to visualize the data coming from a "supportconfig".
+
+When the tool have access to an Uyuni server, then it can run "standalone" or via `mgradm`, allowing to resuse existing Prometheus and Grafana instances.
```

Comment:
```suggestion
When the tool have access to an Uyuni server, then it can run "standalone" or via `mgradm`, allowing to reuse existing Prometheus and Grafana instances.
```